src/Features/Core/Portable/Completion/Providers/ImportCompletionProvider/AbstractExtensionMethodImportCompletionProvider.cs

```diff
@@ -0,0 +1,96 @@
+    internal abstract class AbstractExtensionMethodImportCompletionProvider : AbstractImportCompletionProvider
+    {
+        protected abstract string GenericSuffix { get; }
+
+        protected override bool ShouldProvideCompletion(Document document, SyntaxContext syntaxContext)
+            => syntaxContext.IsRightOfNameSeparator && IsAddingImportsSupported(document);
+
+        protected async override Task AddCompletionItemsAsync(
+            CompletionContext completionContext,
+            SyntaxContext syntaxContext,
+            HashSet<string> namespaceInScope,
+            bool isExpandedCompletion,
+            CancellationToken cancellationToken)
+        {
+            var syntaxFacts = completionContext.Document.Project.LanguageServices.GetRequiredService<ISyntaxFactsService>();
+            if (TryGetReceiverTypeSymbol(syntaxContext, syntaxFacts, out var receiverTypeSymbol))
+            {
+                var items = await ExtensionMethodImportCompletionService.GetUnimportExtensionMethodsAsync(
+                    completionContext.Document,
+                    completionContext.Position,
+                    receiverTypeSymbol,
+                    namespaceInScope,
+                    isExpandedCompletion,
+                    cancellationToken).ConfigureAwait(false);
+
+                completionContext.AddItems(items.Select(Convert));
+            }
+        }
+
+        private static bool TryGetReceiverTypeSymbol(SyntaxContext syntaxContext, ISyntaxFactsService syntaxFacts, [NotNullWhen(true)] out ITypeSymbol? receiverTypeSymbol)
+        {
+            var parentNode = syntaxContext.TargetToken.Parent;
+            var expressionNode = syntaxFacts.GetLeftSideOfDot(parentNode, allowImplicitTarget: true);
+
+            if (expressionNode == null)
+            {
+                receiverTypeSymbol = null;
+                return false;
+            }
+
+            receiverTypeSymbol = syntaxContext.SemanticModel.GetTypeInfo(expressionNode).Type;
+
+            if (receiverTypeSymbol is IErrorTypeSymbol errorTypeSymbol)
+            {
+                receiverTypeSymbol = errorTypeSymbol.CandidateSymbols.Select(s => GetSymbolType(s)).FirstOrDefault(s => s != null);
+            }
+
+            return receiverTypeSymbol != null;
+        }
+
+        private static ITypeSymbol? GetSymbolType(ISymbol symbol)
+        {
+            switch (symbol)
+            {
+                case ILocalSymbol localSymbol:
+                    return localSymbol.Type;
+                case IFieldSymbol fieldSymbol:
+                    return fieldSymbol.Type;
+                case IPropertySymbol propertySymbol:
+                    return propertySymbol.Type;
+                case IParameterSymbol parameterSymbol:
+                    return parameterSymbol.Type;
+                case IAliasSymbol aliasSymbol:
+                    return aliasSymbol.Target as ITypeSymbol;
+            }
+
+            return symbol as ITypeSymbol;
```

Comment:
nit: consider switch expression for all of this.

Reply:
Fixed